StellarDotnetSdk/Requests/DefaultHttpClient.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Reflection;
+
+namespace StellarDotnetSdk.Requests;
+
+internal class DefaultHttpClient : HttpClient
+{
+    /// <summary>
+    ///     Creates an HTTP client with some default request headers and the given bearer token.
+    /// </summary>
+    /// <param name="bearerToken">Bearer token in case the server requires it.</param>
+    public DefaultHttpClient(string? bearerToken = null)
```

Comment:
Maybe we can be more specific and rename it to `DefaultStellarSdkHttpClient` to not confuse people that want to use a `HttpClient` and think ahh there is a `DefaultHttpClient`.

Reply:
It wouldn't be the case since I've set the class as internal. Do think we should make it public?